libs/wallet/src/connectors/injected-connector.ts

```diff
@@ -48,14 +55,34 @@ export const InjectedConnectorErrors = {
 };
 
 export class InjectedConnector implements VegaConnector {
+  isConnected = false;
+  chainId: string | null = null;
   description = 'Connects using the Vega wallet browser extension';
 
+  /**
+   * @deprecated
+   */
   async getChainId() {
     return window.vega.getChainId();
   }
 
-  connectWallet() {
-    return window.vega.connectWallet();
+  async connectWallet(chainId: string) {
+    this.chainId = chainId;
+    try {
+      await window.vega.connectWallet({ chainId });
+      this.isConnected = true;
+    } catch {
+      throw new Error(
+        `could not connect to the vega wallet on chain: ${chainId}`
+      );
+    }
+  }
+
+  onDisconnect(cb: () => void) {
+    window.vega.on('disconnect', () => {
```

Comment:
I think this is the event name 😄 

```suggestion
    window.vega.on('client.disconnect', () => {
```

Reply:
https://github.com/vegaprotocol/vegawallet-browser/pull/667

Reply:
it's `client.disconnected` :D